src/components/VerticalNavigation/VerticalNavigation.js

```diff
@@ -0,0 +1,252 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import ListGroup from '../ListGroup/ListGroup';
+import VerticalNavigationItem from './VerticalNavigationItem';
+
+// TODO compare with
+// http://www.patternfly.org/pattern-library/navigation/vertical-navigation/
+// TODO fix errors and get it working on storybook
+
+/**
+ * VerticalNavigation - The Vertical Navigation pattern
+ */
+export default class VerticalNavigation extends React.Component {
+  // NOTE: For convenience, ALL PROPS of VerticalNavigation will be passed to all VerticalNavigationItem children.
+  // EXCEPTIONS:
+  // - The children prop, for obvious reasons.
+  // - The depth prop will automatically be set to the currently nested depth (primary, secondary or tertiary).
+  // - Any props passed explicitly to VerticalNavigationItem children will take precendence over inherited props.
+  static renderChildren(props) {
+    const { children, ...parentProps } = props; // TODO maybe take specific props instead of passing all of them?
+    const depth = VerticalNavigation.getNextDepth(props.depth || 'primary');
+    return React.Children.map(children, child => {
+      if (child.type === VerticalNavigationItem) {
+        return React.cloneElement(child, {
+          ...parentProps,
+          depth,
+          ...child.props, // We include this to allow overriding these props inline on a VerticalNavigationItem.
+        });
+      } else {
+        return child;
+      }
+    });
+  }
+
+  static getNextDepth(depth) {
+    return depth === 'primary' ? 'secondary' : 'tertiary';
+  }
+
+  constructor() {
+    super();
+    this.state = {
+      hoverSecondaryNav: false,
+      hoverTertiaryNav: false,
+      collapsedSecondaryNav: false,
+      collapsedTertiaryNav: false,
+    };
+    this.handleNavBarToggleClick = this.handleNavBarToggleClick.bind(this);
+  }
+
+  handleNavBarToggleClick() {
+    const { hideTopBanner } = this.state;
+    this.setState({
+      hideTopBanner: !hideTopBanner,
+    });
+  }
+
+  handlePrimaryClick(item) {
+    // TODO
+  }
+
+  handlePrimaryHover(item) {
+    // TODO
+  }
+
+  handlePrimaryBlur(item) {
+    // TODO
+  }
+
+  handleSecondaryClick(item) {
+    // TODO
+  }
+
+  handleSecondaryHover(item) {
+    // TODO
+  }
+
+  handleSecondaryBlur(item) {
+    // TODO
+  }
+
+  handleTertiaryClick(item) {
+    // TODO
+  }
+
+  handleTertiaryHover(item) {
+    // TODO
+  }
+
+  handleTertiaryBlur(item) {
+    // TODO
+  }
+
+  render() {
+    const {
+      items,
+      hidden,
+      persistentSecondary,
+      pinnableMenus,
+      hiddenIcons,
+      brandSrc,
+      brandAlt,
+      showBadges,
+      showMobileNav,
+      showMobileSecondary,
+      showMobileTertiary,
+      forceHidden,
+      hideTopBanner,
+      navCollapsed,
+      inMobileState,
+      activeSecondary,
+      children,
+      topBannerContents,
+      notificationDrawer /* TODO notification drawer components? */,
```

Comment:
I would leave this out and allow the application to add it separately. The drawer trigger should be added to the topBannerContents and notification drawer would be part of the app content.

Reply:
Well, that is sort of what I'm doing here already. The notification drawer is a child of the top banner, so it has to be rendered by this component (React doesn't allow external DOM modifications). So we just accept a prop `notificationDrawer` that is just the entire node tree to be rendered in that container.  The application can pass a whole other component as that prop.

Reply:
I don't believe notificationDrawer needs to be a child of the top banner.

Reply:
If it is part of the vertical navigation, it does buy you the correct top positioning and height otherwise it is not necessary, the application would then need to specify those values. We could handle that in the notification drawer component.

Just seems better to keep them separated to me. I'm OK either way though.

Reply:
I see. Okay. It was a child of the top banner in the pf-ng code and the PF core example markup. If we don't do that, I feel like we should make that change upstream maybe?

Reply:
there is no upstream change necessary though. You can have it either way. There is just the preset position and height setting gained by having the drawer-pf a child of navbar-pf-vertical